### ISSUE TITLE
Temporary unhealthy containers during auto backups

### ISSUE THREAD

**Issue by frederikb96** [nextcloud/all-in-one#1611]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Make some big changes to your files (such that the backup runs for at least some minutes)
1. Via the Nextcloud AIO interface, schedule an auto backup for in 2 minutes
2. Open portainer or the command line, and observer the containers
3. When the backup starts, all containers are stopped, the backup container starts and all other containers start

### Expected behavior 
All containers directly go into the healthy state

### Actual behavior 
The Nextcloud container and the Apache container goes into the unhealthy state and afterwards go into the healthy state

### My guess
If I manually do a backup, I manually have to start the containers again, and every container goes directly into the healthy state.
I guess, since the containers are already somehow started in the auto backup process with the backup container and the real start is just postponed, they run into some timeout and go into the unhealthy state. This is annoying, since I get notifications from netdata, that there are unhealthy containers, every night. I would rather not adjust the netdata settings, since I really want to be warned if there are unhealthy containers.

I guess if the backup is really small, the containers start fast enough, such that no unhealthy state is triggered.

### Host OS 
Ubuntu 22.04 server

#### Nextcloud AIO version 
v4.0.1

#### Current channel 
Beta

**Comment by Zoey2936** [comment]
The Container reports unhealthy if the services inside the container don't bind their ports, sometimes the containers need to wait for other containers to start and then bind their ports and because of this theoretically the containers are unhealthy. I don't see an option to differentiate between a container that takes long to start and a really unhealthy container.
The only Thing I see we could do is to increase the start period (https://docs.docker.com/engine/reference/builder/#healthcheck)

**Comment by szaimen** [comment]
> I would rather not adjust the netdata settings, since I really want to be warned if there are unhealthy containers.

I would rather suggest checking if one of the containers is restarting which is also what we do in AIO and the better approach, imho.

> I don't see an option to differentiate between a container that takes long to start and a really unhealthy container.
> The only Thing I see we could do is to increase the start period (https://docs.docker.com/engine/reference/builder/#healthcheck)

Yes, I also dont see a better way than adjusting the start period a bit. However this will variy depending on available hardware, internet speed, etc. 

Anyway, @Zoey2936 could you maybe create a PR that increases the period for Apache and Nextcloud? :)

**Comment by Zoey2936** [comment]
> Anyway, @Zoey2936 could you maybe create a PR that increases the period for Apache and Nextcloud? :)

How many seconds would you suggest?

**Comment by szaimen** [comment]
> How many seconds would you suggest?

Lets ask @frederikb96 how long it takes for him to start these containers...

**Comment by frederikb96** [comment]
Thanks for the replies!
I still do not understand the difference between the process of doing an auto backup and the process of doing a manual backup and after it is done, starting the containers again.

In the first case, I get unhealthy containers. In the second, I don't.
Normally, my containers start fast enough.

My assumption is that for the auto case, the containers take as long to start as the backup takes to run. See this log from last night. The backup took ~15 minutes and consequently, the containers took 15 minutes to start. Can you not just start the containers, once the backup is done, like in the manual backup process? Or do I miss something?

![image](https://user-images.githubusercontent.com/83548283/209822690-3c9a8727-e7c8-4d58-ac4c-3c1f6debae16.png)

**Comment by szaimen** [comment]
> My assumption is that for the auto case, the containers take as long to start as the backup takes to run. See this log from last night. The backup took ~15 minutes and consequently, the containers took 15 minutes to start

Ah I see. Yes that makes sense.

We don't do it like proposed since that would block the AIO interface a long time and you would not get any feedback what is happening. So that is actually a UX decision. In that case we would probably need to increase the timeout to 15min or only start the containers after the backup is done. However as I said would this block the AIO interface potentially for a long time which is neither desired .

So no good idea how to solve both problems at the same time...

**Comment by szaimen** [comment]
I've created a POC how this could maybe be fixed: https://github.com/nextcloud/all-in-one/pull/1615

**Comment by frederikb96** [comment]
Ah ok, I see, thanks :)

**Comment by szaimen** [comment]
This is now released with v4.1.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel
